modules/apps/roles/roles-admin-web/src/main/resources/META-INF/resources/edit_role_permissions_summary.jsp

```diff
@@ -244,14 +244,21 @@
 			searchContainer="<%= searchContainer %>"
 			searchResultCssClass="show-quick-actions-on-hover table table-autofit"
 		/>
-
-		<c:if test="<%= searchContainer.getTotal() > 0 %>">
-			<div class="autofit-col sheet-footer taglib-search-iterator-page-iterator-bottom">
-				<liferay-ui:search-paginator
-					markupView="lexicon"
-					searchContainer="<%= searchContainer %>"
-				/>
-			</div>
-		</c:if>
-	</div>
-</div>
\ No newline at end of file
+	</clay:sheet-section>
+
+	<c:if test="<%= searchContainer.getTotal() > 0 %>">
+		<clay:sheet-footer>
+			<clay:content-row>
+				<clay:content-col
+					className="taglib-search-iterator-page-iterator-bottom"
+					expand="true"
+				>
+					<liferay-ui:search-paginator
+						markupView="lexicon"
+						searchContainer="<%= searchContainer %>"
+					/>
+				</clay:content-col>
+			</clay:content-row>
+		</clay:sheet-footer>
+	</c:if>
```

Comment:
I tried to move this section outside the `</clay:sheet-section>` but it's not compiling. If I keep the`<div class="sheet-section"></div>` it works. It seems to be some sort of variables reset triggered by `<clay:sheet-footer>`

---
````
Found gradlew at /Users/eduardoallegrini/projects/liferay-portal/liferay-portal
Configuration on demand is an incubating feature.
> Task :apps:roles:roles-admin-web:cleanBaseline UP-TO-DATE
> Task :apps:roles:roles-admin-web:cleanBuildBundleDir UP-TO-DATE
> Task :apps:roles:roles-admin-web:cleanBuildCSS
> Task :apps:roles:roles-admin-web:cleanBuildEnvironment UP-TO-DATE
> Task :apps:roles:roles-admin-web:cleanCompileJSP
> Task :apps:roles:roles-admin-web:cleanCompileJava
> Task :apps:roles:roles-admin-web:cleanCompileTestIntegrationJava UP-TO-DATE
> Task :apps:roles:roles-admin-web:cleanCompileTestJava UP-TO-DATE
> Task :apps:roles:roles-admin-web:cleanCopyCSS
> Task :apps:roles:roles-admin-web:cleanCopyLibs UP-TO-DATE
> Task :apps:roles:roles-admin-web:cleanCopyTLDDocResources UP-TO-DATE
> Task :apps:roles:roles-admin-web:cleanDependencies UP-TO-DATE
> Task :apps:roles:roles-admin-web:cleanDownloadNode
> Task :apps:roles:roles-admin-web:cleanEclipseJdt UP-TO-DATE
> Task :apps:roles:roles-admin-web:cleanGenerateJSPJava
> Task :apps:roles:roles-admin-web:cleanJar
> Task :apps:roles:roles-admin-web:cleanJarCompileIncludeFragment UP-TO-DATE
> Task :apps:roles:roles-admin-web:cleanJarJSDoc UP-TO-DATE
> Task :apps:roles:roles-admin-web:cleanJarJSP UP-TO-DATE
> Task :apps:roles:roles-admin-web:cleanJarJavadoc UP-TO-DATE
> Task :apps:roles:roles-admin-web:cleanJarSources UP-TO-DATE
> Task :apps:roles:roles-admin-web:cleanJarSourcesCommercial UP-TO-DATE
> Task :apps:roles:roles-admin-web:cleanJarTLDDoc UP-TO-DATE
> Task :apps:roles:roles-admin-web:cleanJavadoc UP-TO-DATE
> Task :apps:roles:roles-admin-web:cleanJsdoc UP-TO-DATE
> Task :apps:roles:roles-admin-web:cleanMergeArtifactsPublishCommands UP-TO-DATE
> Task :apps:roles:roles-admin-web:cleanMergeLang UP-TO-DATE
> Task :apps:roles:roles-admin-web:cleanPackageRunBuild
> Task :apps:roles:roles-admin-web:cleanPmdMain UP-TO-DATE
> Task :apps:roles:roles-admin-web:cleanPmdTest UP-TO-DATE
> Task :apps:roles:roles-admin-web:cleanPmdTestIntegration UP-TO-DATE
> Task :apps:roles:roles-admin-web:cleanProcessResources UP-TO-DATE
> Task :apps:roles:roles-admin-web:cleanProcessTestIntegrationResources UP-TO-DATE
> Task :apps:roles:roles-admin-web:cleanProcessTestResources UP-TO-DATE
> Task :apps:roles:roles-admin-web:cleanProjects UP-TO-DATE
> Task :apps:roles:roles-admin-web:cleanProperties UP-TO-DATE
> Task :apps:roles:roles-admin-web:cleanSpotbugsMain UP-TO-DATE
> Task :apps:roles:roles-admin-web:cleanSpotbugsTest UP-TO-DATE
> Task :apps:roles:roles-admin-web:cleanSpotbugsTestIntegration UP-TO-DATE
> Task :apps:roles:roles-admin-web:cleanTasks UP-TO-DATE
> Task :apps:roles:roles-admin-web:cleanTest UP-TO-DATE
> Task :apps:roles:roles-admin-web:cleanTestIntegration UP-TO-DATE
> Task :apps:roles:roles-admin-web:cleanTlddoc UP-TO-DATE
> Task :apps:roles:roles-admin-web:cleanWatch UP-TO-DATE
> Task :apps:roles:roles-admin-web:cleanWriteFindBugsProject UP-TO-DATE
> Task :apps:roles:roles-admin-web:clean

> Task :apps:roles:roles-admin-web:compileJava
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

> Task :apps:roles:roles-admin-web:copyLibs SKIPPED
> Task :apps:roles:roles-admin-web:copyCSS

> Task :apps:roles:roles-admin-web:buildCSS
Using native Sass compiler
Parsed /META-INF/resources/css/main.scss in 538ms

> Task :downloadNode FROM-CACHE
> Task :apps:roles:roles-admin-web:downloadNode SKIPPED
> Task :apps:roles:roles-admin-web:expandJSCompileDependencies UP-TO-DATE
> Task :apps:roles:roles-admin-web:npmInstall SKIPPED

> Task :yarnInstall
Installing the Node.js packages

> Task :yarnInstallModules
yarn install v1.13.0
[1/4] Resolving packages...
success Already up-to-date.
Done in 1.57s.

> Task :apps:roles:roles-admin-web:packageRunBuild FROM-CACHE
> Task :apps:roles:roles-admin-web:processResources
> Task :apps:roles:roles-admin-web:replaceSoyTranslation SKIPPED
> Task :apps:roles:roles-admin-web:wrapSoyAlloyTemplate SKIPPED
> Task :apps:roles:roles-admin-web:mergeLang

> Task :apps:roles:roles-admin-web:generateJSPJava
Compiling JSP files of project ':apps:roles:roles-admin-web' with /Users/eduardoallegrini/projects/liferay-portal/bundles/tomcat-9.0.33/webapps/ROOT/WEB-INF/lib/util-taglib.jar as dependency in place of 'com.liferay.portal:com.liferay.util.taglib'

> Task :apps:roles:roles-admin-web:classes

> Task :apps:roles:roles-admin-web:compileJSP FAILED
/Users/eduardoallegrini/projects/liferay-portal/liferay-portal/modules/apps/roles/roles-admin-web/build/jspc/org/apache/jsp/edit_005frole_005fpermissions_005fsummary_jsp.java:698: error: cannot find symbol
        _jspx_th_c_if_0.setTest( searchContainer.getTotal() > 0 );
                                 ^
  symbol:   variable searchContainer
  location: class edit_005frole_005fpermissions_005fsummary_jsp
/Users/eduardoallegrini/projects/liferay-portal/liferay-portal/modules/apps/roles/roles-admin-web/build/jspc/org/apache/jsp/edit_005frole_005fpermissions_005fsummary_jsp.java:731: error: cannot find symbol
                  _jspx_th_liferay$1ui_search$1paginator_0.setSearchContainer( searchContainer );
                                                                               ^
  symbol:   variable searchContainer
  location: class edit_005frole_005fpermissions_005fsummary_jsp
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
2 errors


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':apps:roles:roles-admin-web:compileJSP'.
> Compilation failed; see the compiler error output for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 16s
57 actionable tasks: 18 executed, 2 from cache, 37 up-to-date
````

Reply:
We are not sure if this is a standard by using taglibs.

Another thing we have decided is to include a structure of `footer-sheet > content-row > content-col` instead of a mix of classes `sheet-footer autofit-col`. in order to make the use of new grids more clear and respectful

Reply:
Tag execution will create a new Java context. So:

```jsp
<%
String a = "a"
%>

<a:foo>

    <% 
    String b = "b";
    %>

    <a:bar>
       <%= a %> OK!
       <%= b %> OK!
    </a:bar>
</a:foo>

<%= b %> NOT OK!
```

If you have a scriptlet `<% ... %>` generating Java variables that you need to access elsewhere in the jsp, you need to make sure those are not nested inside tags.